cpp/src/groupby/hash/groupby.cu

```diff
@@ -307,6 +437,79 @@ void compute_single_pass_aggs(table_view const& keys,
   }
 }
 
+template <bool keys_have_nulls, typename Map>
+void compute_second_pass_aggs(table_view const& keys,
```

Comment:
This looks really complicated. What are "second pass aggs"?

Reply:
`compute_single_pass_aggs` is old function name for simple aggregations calculation. 
following that, this one is named as `compute_second_pass_aggs` (or it could be `compute_double_pass_aggs`).

Reply:
I see what's going on now. You're trying to do _all_ compound aggregations in a single pass. I think that's a lot more complication than what is needed. We should just handle the "second pass" one by one in the `finalize` step.

Reply:
I implemented like that first. But it will face performance hit when number of columns are high. So, switched to this approach.

Reply:
Here's a high level description of what I think this should look like:

- Preprocess and decompose original compound aggs to simple aggs
- Compute all simple aggs in a single pass
- For each compound agg, invoke `finalize` which will compute the final result from the computed simple aggs

Reply:
@jrhemstad  For the third point,
> it will face performance hit when number of columns are high

because `finalize` will be called for each compound aggregation and for each column. If MEAN of N columns is requested, it will launch N kernel launches!

Reply:
Yes, that is expected. I don't expect that to be a problem and it seems like premature optimization to try and do anything about it at this point.

Reply:
updated to individual finalize()